internal/service/streamprocessor/resource.go

```diff
@@ -187,6 +195,7 @@ func (r *streamProcessorRS) Update(ctx context.Context, req resource.UpdateReque
 		).Execute()
 		if err != nil {
 			resp.Diagnostics.AddError("Error stopping stream processor", err.Error())
+			return
```

Comment:
would be nice to have some linter or something to detect when we're not returning after adding error

Reply:
An alternative is creating a helper function that could be called with

```golang
if err != nil {
  return add_error_message(resp, err, "Error stopping stream_processor")
  }
```
It wouldn't return anything but makes sure the error is added and can be called with `return` directly to avoid an extra `return` statement on the next line. WDYT?

Reply:
that could work.   one pro about having param diagnostics *diag.Diagnostics is that you can potentially add multiple errrors in the same function,  and it's quite direct from RS functions to do if diagnostics.HasError()

Reply:
Good idea! I'll keep this PR as is, but I will add it to the model conversion PR!